components/config-mgmt-service/backend/elastic/suggestions.go

```diff
@@ -177,19 +178,18 @@ func (es Backend) getArrayAggSuggestions(term string, text string, filters map[s
 	typeQuery := elastic.NewTypeQuery(IndexNodeState)
 	filters["exists"] = []string{"true"}
 	boolQuery := newBoolQueryFromFilters(filters)
-	textLower := strings.ToLower(text)
 
 	boolQuery = boolQuery.Must(typeQuery)
 
 	// return all unless text has at least 2 chars
 	if len(text) >= 2 {
-		matchQuery := elastic.NewMatchQuery(fmt.Sprintf("%s.engram", term), textLower)
+		matchQuery := elastic.NewMatchQuery(fmt.Sprintf("%s.engram", term), text)
 		boolQuery = boolQuery.Must(matchQuery)
 	}
 	// multiplying the size by 10 as elasticsearch will sort array aggregations by doc_count. Will trim it back to size once we match it again in go
 	aggs := elastic.NewTermsAggregation().Field(term).Size(SuggestionQuerySize)
 	if len(text) >= 2 {
-		aggs = aggs.Include(".*" + textLower + ".*")
+		aggs = aggs.Include(createCaseInsensitivePattern(text))
```

Comment:
We are filtering the array of values with a case-insensitive regex. Before all the values in the suggestion were converted to lower case so case-insensitive matching was not needed.